app/api/public_v2.py

```diff
@@ -0,0 +1,289 @@
+        if week_ago_day in self.key_to_date[state]:
+            data_for_week_ago = self.key_to_date[state][week_ago_day]
+            field_value_for_week_ago = getattr(data_for_week_ago, field_name)
+            if field_value_for_day is not None and \
+                    field_value_for_week_ago is not None and \
+                    field_value_for_week_ago > 0:
+                pct_change = (field_value_for_day-field_value_for_week_ago)/field_value_for_week_ago
+                return round(pct_change * 100, 1)
+
+        return None
+
+    def seven_day_average(self, core_data, field_name):
+        field_value_for_day = getattr(core_data, field_name)
+        if field_value_for_day is None:
+            return None
+
+        state = getattr(core_data, 'state') or 'US'
+        seven_day_values = []
+        for i in range(7):
+            some_prior_date = core_data.date - timedelta(days=i)
+            if some_prior_date in self.key_to_date[state]:
+                data_for_some_prior_date = self.key_to_date[state][some_prior_date]
+                field_value_for_some_prior_date = getattr(data_for_some_prior_date, field_name)
+                if field_value_for_some_prior_date is not None:
+                    seven_day_values.append(field_value_for_some_prior_date)
+
+        assert len(seven_day_values) <= 7
+        if len(seven_day_values) > 0:
+            return sum(seven_day_values) / len(seven_day_values)
+
+        return None
+
+    def calculate_values(self, core_data, field_name):
+        """
+        Returns calculated values for the given core data field as a dictionary.
+
+        Parameters
+        ----------
+        core_data : CoreData
+            The CoreData object for which we need to calculate a bunch of derived values 
+        field_name : str
+            The CoreData property we're calculating values for, e.g. "positiveTestsViral"
+        """
+        return {
+            'population_percent': None,  # TODO: compute this
+            'change_from_prior_day': self.change_from_prior_day(core_data, field_name),
+            'seven_day_average': self.seven_day_average(core_data, field_name),
+            'seven_day_change_percent': self.seven_day_change_percent(core_data, field_name),
+        }
+
+
+def recursive_tree_to_simple_output(tree, core_data):
+    # walk through it recursively and populate the fields from core_data
+    for k, v in tree.items():
+        # if v is a string, we're at a leaf, need to replace v with the actual value from core_data
+        if isinstance(v, str):
+            tree[k] = getattr(core_data, v)
+        else:
+            # need to recurse one level down
+            recursive_tree_to_simple_output(v, core_data)
+
+
+def convert_core_data_to_simple_output(core_data):
+    core_data_copy = copy.deepcopy(_MAPPING)
+    recursive_tree_to_simple_output(core_data_copy, core_data)
+    return core_data_copy
+
+
+def recursive_tree_to_full_output(tree, core_data, calculator):
```

Comment:
I don't have strong feelings about this, but I'm wondering if it makes sense for `recursive_tree_to_simple_output` and `recursive_tree_to_full_output` to be just one function. They both do the same thing, and the non-simple version just has the extra step of calling the calculator. It seems like unifying them could simplify future changes, maybe?

Reply:
I actually had them as one function first, then split them out because for some reason it was easier to reason about at the time while I was trying to remember how recursion works, but you're right that unifying is better at this point - done :)